scripts/find-ts-version.js

```diff
@@ -1,8 +1,8 @@
 // @ts-check
 const fs = require("fs");
 
-const FIRST_TYPESCRIPT_LINE = /typescript\@\^?(\d+.\d+(.\d+)?):/;
-const TYPESCRIPT_VERSION_LINE = /version "(\d+.\d+)(.\d+)?"/;
+const FIRST_TYPESCRIPT_LINE = /typescript@\S*:/;
+const TYPESCRIPT_VERSION_LINE = /version "(\d+\.\d+\.\d+)"/;
```

Comment:
Originally, `TYPESCRIPT_VERSION_LINE` was used to generate `test/ts-version.ts` with `type TsVersion` which was used in unit tests, when output was different for different TypeScript versions.

I've tidied up the usage just now in https://github.com/ts-essentials/ts-essentials/commit/25b538c27d8917f798a3a87d7003102a32f1d412

But it's still useful in e.g. https://github.com/ts-essentials/ts-essentials/blob/master/test/strict-extract.ts#L53

I still believe that writing `TsVersion extends "4.5"` is easier than `TsVersion extends "4.5.5"`, but to be honest, there is no much difference.

Therefore I suggest:

1. Please leave a JSDoc comment for `TYPESCRIPT_VERSION_LINE` to explain that, currently, there are 2 purposes for this Regexp: it defines `TsVersion` and it is used in conditional tests for a `semver` library

2. Please update `strict-extract.ts` unit test so the CI check is passed for TypeScript 4.5

What do you think?

Reply:
1. Yeah sure, I'll add a comment there.
2. Apologies, but I’m having trouble understanding the need for a conditional test in `strict-extract`. Because both test cases seem to pass in `v4.5` and later versions (playground link: https://tsplay.dev/NV8BMW). So, maybe we can remove the usage of `TsVersion` from `strict-extract` as well.

And maybe in the future we can switch to this if we want to avoid specifying the patch number:
```ts
TsVersion extends `4.5.${number}`
```

Reply:
FYI Your link points to TS 5.0, you need 4.5 - https://tsplay.dev/wOQ9Rm (but it worked).

The issue exists in TS versions:

- 4.6 - https://tsplay.dev/mLVP2w
- 4.7 - https://tsplay.dev/w1DvAW

You will see:

```
Type '{ type: "dog" | "cat"; }' is not assignable to type 'Partial<Mouse>'.
  Types of property 'type' are incompatible.
    Type '"dog" | "cat"' is not assignable to type '"mouse" | undefined'.
      Type '"dog"' is not assignable to type '"mouse"'
```

Example - https://github.com/ts-essentials/ts-essentials/actions/runs/11626949786/job/32379475714

It depends on how `Partial` is implemented, it seems to be broken in versions 4.6 and 4.7

Anyway, I've updated the `StrictExtract` tests in https://github.com/ts-essentials/ts-essentials/commit/dc1cd6979dde65b440ba5e618ca136c5d45642a2

Reply:
Interesting, doesn’t seem to be an issue with `Partial` but rather with how the type constraint is being enforced. Because the following fails `v4.5` and `v4.6`, but works in others.

Playground: https://tsplay.dev/WKQDpN
<img width="1443" alt="image" src="https://github.com/user-attachments/assets/cb38e929-de23-4cdd-8112-134bdce1a96b">

Anyways, I've pulled in your changes and updated the `strict-extract` conditional tests ([commit](https://github.com/ts-essentials/ts-essentials/pull/416/commits/3d54e93416812f155090dcb2528d53cbe13f09d6)). LGTM now!

Reply:
Great, thank you!